draft-ietf-tls-tls13.md

```diff
@@ -2835,19 +2810,25 @@ message is sent as the last message before the CertificateVerify.
 Structure of this Message:
 
 %%% Hello Messages
+          enum { (65535) } ConfigurationExtensionType;
+
+          struct {
+              ConfigurationExtensionType extension_type;
+              opaque extension_data<0..2^16-1>;
+          } ConfigurationExtension;
 
           struct {
               opaque configuration_id<1..2^16-1>;
               uint32 expiration_date;
               NamedGroup group;
               opaque server_key<1..2^16-1>;
-              Boolean early_data_allowed;
+              EarlyDataType early_data_type;
```

Comment:
You should remove the "Primitive Types" section now that you're not using this boolean type anymore.

I also suggest "allowed_early_data_types", just for clarity.